File: fake_daemon.dart
import 'dart:convert';
import 'dart:io';

Future<void> main() async {
  print('Flutter Daemon started!');
  stdin.transform(utf8.decoder).listen(
    (input) {
      print('Daemon was sent: $input');
    },
    onDone: () {
      print('Stream done! WERE YOU EXPECTING THIS?');
    },
    onError: (e, s) {
      print('ERROR! $e\n$s');
    },
  );

  // Keep the process alive (see additional notes in readme).
  await Future.delayed(const Duration(hours: 1));
}

File: fake_daemon.js
console.log('Flutter Daemon started!');

process.stdin.on('data', function (input) {
  console.log(`Daemon was sent: ${input}`);
});

process.stdin.on('end', function () {
  console.log('Stream done! WERE YOU EXPECTING THIS?');
});

process.stdin.on('error', function (e) {
  console.log(`Error! ${e}`);
});
